x/oracle/keeper/historic_price.go

```diff
@@ -163,6 +253,25 @@ func (k Keeper) getHistoricPrices(
 	return historicPrices
 }
 
+// getMedians returns all the medians of a given denom in a given block
+// range.
+func (k Keeper) getMedians(
+	ctx sdk.Context,
+	denom string,
+	startBlock uint64,
+	endBlock uint64,
+) []sdk.Dec {
+	medians := []sdk.Dec{}
+	k.IterateMedians(ctx, denom, func(exchangeRate sdk.Dec, block uint64) bool {
+		if block >= startBlock && block <= endBlock {
```

Comment:
`startBlock` should be included in the key prefix search, to avoid traversing and deserializing not needed records.

Reply:
This is tricky with the KVStore iterator since the prefix is changing every iteration. I changed it to not use the iterator and accomplish the same result in `getMedians()`. Let me know what you think about that solution.